terraform/tfschema/customtype.go

```diff
@@ -0,0 +1,106 @@
+package tfschema
+
+import (
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/types/wrappers"
+	schema "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// SchematypesBoolOption represents schema of custom bool value with true default
+func SchematypesBoolOption() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeBool,
+		Optional: true,
+	}
+}
+
+// SchemagithubcomgravitationalteleportapitypeswrappersTraits represents traits schema
+func SchemagithubcomgravitationalteleportapitypeswrappersTraits() *schema.Schema {
```

Comment:
Is this a generated function name / does it have to look like this? If we can edit it, I'd at least make inner words CamelCase so eyes don't bleed reading this :) If not, disregard.

Reply:
It is generated, yes :-) I actually had the question about it. I've used lower case letter as a package designator, and upper case as a type designator (as usual). I were afraid that if all components were camel cased, there could be duplication: for instance, type WrappersFoo would have the same method name as wrappers.Foo (WrappersFoo). I thought that in future we could provide replacement dictionary on generator side to avoid eye bleeding :-) Or, specifically here underscore could be used as a component separator - but such method names would be considered invalid by the linter.